internal/cache/controller_obj_cache.go

```diff
@@ -2523,37 +2527,66 @@ func checkRequiredValuesYaml() bool {
 // validateAndConfigureSeGroup validates SeGroup configuration provided during installation
 // and configures labels on the SeGroup if not present already
 func validateAndConfigureSeGroup(client *clients.AviClient) bool {
-	// Not applicable for NodePort mode / disable route is set as True
+	// Indulge in Marker based SEGroup searching only when user input is empty.
+	// SE Group selection priority
+	// 1. User input
+	seGroupToUse := lib.GetSEGNameEnv()
+
+	// TODO: pagination
+	uri := "/api/serviceenginegroup/?include_name&page_size=100&cloud_ref.name=" + utils.CloudName
+	elems := []json.RawMessage{}
+
+	// 2. Marker based (only advancedL4)
+	if seGroupToUse == "" && lib.GetAdvancedL4() {
```

Comment:
If we are checking seGroupToUse == "", then we can return before this if seGroupToUse != ""